### PR TITLE
inital quick review

### PR DESCRIPTION
**Pull request by TychonautVII**
Opening this pull request just a means to provide a couple of high level comments, mostly about inputs and outputs. You don't have to merge this if you don't want to!

### CODE REVIEW

File: Cropland_Nutrient and Tillage Only.R [hunk 1]
```diff
@@ -47,38 +47,64 @@
...
+
+# Libraries
+library(jsonlite)
+library(tidyverse)
+library(parallel)
+library(future.apply)
+library(rstudioapi)
+# it be good to track the specific versions of these library you're using, 
+# either manually in the readme or using something like ren if you're familiar 
+# I couldn't run this at first because I didn't see the plyr dependency for instance
```
Comment (TychonautVII) [inline Cropland_Nutrient and Tillage Only.R:59]: Tracking libraries

File: Cropland_Nutrient and Tillage Only.R [hunk 2]
```diff
@@ -47,38 +47,64 @@
...
+        
+            # I don't think you should need this file, because this information 
+            # is already in croplandex.json in the aoi_subregions. 
+            # The other input file you're getting according to  
+            # https://mathematicampr.atlassian.net/wiki/spaces/AFOLU/pages/3212444049/Code+requirements
+            # is land cover information which isn't in croplandex.json 
             gee<-jsonlite::fromJSON('gee_ex.json',flatten=TRUE)
             geedf<-as.data.frame(gee)
             colnames(geedf)<-names(gee$aoi_subregions)
             geedf$aoi_id<-c(1:nrow(geedf))
```
Comment (TychonautVII) [inline Cropland_Nutrient and Tillage Only.R:76]: Inputs

File: Cropland_Nutrient and Tillage Only.R [hunk 3]
```diff
@@ -101,6 +127,11 @@
             df<-df[,-grep("uncertainty|error",names(df))]
       #Parameters
             AOIs<-unique(df$aoi_id)
+            
+            
+            # it be good to make paramateters like this that we might theoretically want to change 
+            # more obvious, maybe bringing them to the top of the file? 
+            # Just to distinquish them from the methodology driven numbers. 
             nx=500 #Number of Monte Carlo iterations
```
Comment (TychonautVII) [inline Cropland_Nutrient and Tillage Only.R:126]: Paramaters

File: Cropland_Nutrient and Tillage Only.R [hunk 4]
```diff
@@ -185,6 +216,13 @@
...
         nest_legacy(.key = "Var")
-      json_output<-jsonlite::toJSON(resdf)
-      write(json_output, paste0(interv_sub,".json"))
+      
+      # this doesn't seem technically consistent with 
+      # https://mathematicampr.atlassian.net/wiki/spaces/AFOLU/pages/3212444049/Code+requirements#Expected-output
+      # but that might be alright if it's cool with Jenny? 
+      # We need an agreed upon output format for your code though to give to the app team 
+      # so we should update this or the confluence so they're consistent
+
```
Comment (TychonautVII) [inline Cropland_Nutrient and Tillage Only.R:216]: Output definition

File: Cropland_Nutrient and Tillage Only.R [hunk 5]
```diff
@@ -47,38 +47,64 @@
...
+# it be good to track the specific versions of these library you're using, 
+# either manually in the readme or using something like ren if you're familiar 
+# I couldn't run this at first because I didn't see the plyr dependency for instance
+
+
     # Extract data from json input
-        setwd("C:/Users/lme38/Dropbox/Soil Simplified/Ecodit")
+        #trying to still set the working directory but more generically, do you think this is okay
+        current_path = rstudioapi::getActiveDocumentContext()$path 
+        setwd(dirname(current_path ))
```
Comment (TychonautVII) [inline Cropland_Nutrient and Tillage Only.R:65]: Removing path

File: Cropland_Nutrient and Tillage Only.R [hunk 6]
```diff
@@ -185,6 +216,13 @@
...
-      write(json_output, paste0(interv_sub,".json"))
+      
+      # this doesn't seem technically consistent with 
+      # https://mathematicampr.atlassian.net/wiki/spaces/AFOLU/pages/3212444049/Code+requirements#Expected-output
+      # but that might be alright if it's cool with Jenny? 
+      # We need an agreed upon output format for your code though to give to the app team 
+      # so we should update this or the confluence so they're consistent
+      
+      json_output<-jsonlite::toJSON(resdf, pretty=TRUE)
+      write(json_output, paste0("output/",interv_sub,".json"))
```
Comment (TychonautVII) [inline Cropland_Nutrient and Tillage Only.R:218]: set pretty to true and changed output dir

**Review by lisaeash — APPROVED** [review]

**Pull request merged**